dotcom-rendering/playwright/tests/parallel-4/atom.interactivity.e2e.spec.ts

```diff
@@ -0,0 +1,215 @@
+			await loadPage(page, `/Article/${url}`);
+			await page.locator(`[data-snippet-type=${type}]`).click();
+			await page.locator('[data-testid="dislike"]').click();
+			await expectToBeVisible(page, '[data-testid="feedback"]');
+		});
+	});
+}
+
+for (const { type, url } of genericTests) {
+	test.describe(type, () => {
+		test('should render', async ({ context, page }) => {
+			await disableCMP(context);
+			await loadPage(page, `/Article/${url}`);
+			await expectToBeVisible(page, `[data-snippet-type=${type}]`);
+		});
+	});
+}
+
+test.describe('Why do wombats do square poos?', () => {
+	test('when I get the answer wrong, it should display the right answer when I click Reveal', async ({
+		context,
+		page,
+	}) => {
+		await disableCMP(context);
+		await loadPage(page, `/Article/${quizAtomUrl}`);
+
+		await waitForIsland(page, 'KnowledgeQuizAtom', {
+			status: 'hydrated',
+		});
+
+		// Establish that the elements showing the results are not present
+		await expectLocatorToNotExist(
+			page,
+			page
+				.locator('[data-atom-type=knowledgequiz] fieldset')
+				.first()
+				.filter({
+					has: page.locator(
+						'[data-answer-type=non-selected-correct-answer]',
+					),
+				}),
+		);
+		await expectLocatorToNotExist(
+			page,
+			page
+				.locator('[data-atom-type=knowledgequiz] fieldset')
+				.first()
+				.filter({
+					has: page.locator(
+						'[data-answer-type=non-incorrect-answer]',
+					),
+				}),
+		);
+
+		// Click an incorrect answer
+		await page.getByText('Because they have square bottoms').click();
+		// Click Reveal to show the results
+		await page
+			.locator('[data-atom-type=knowledgequiz]')
+			.getByText('Reveal')
+			.first()
+			.click();
+
+		// We got the question wrong!
+		// Our choice is shown as wrong (red) and the actual correct answer is shown in green
+		await expectToExist(page, '[data-answer-type=incorrect-answer]');
+		await expectToExist(
+			page,
+			'[data-answer-type=non-selected-correct-answer]',
+		);
+	});
+
+	test('when I get the answer right, it should commend my skills when I click Reveal', async ({
+		context,
+		page,
+	}) => {
+		await disableCMP(context);
+		await loadPage(page, `/Article/${quizAtomUrl}`);
+
+		await waitForIsland(page, 'KnowledgeQuizAtom', {
+			status: 'hydrated',
+		});
+
+		// Establish that the elements showing the results are not present
+		await expectLocatorToNotExist(
+			page,
+			page
+				.locator('[data-atom-type=knowledgequiz] fieldset')
+				.first()
+				.filter({
+					has: page.locator(
+						'[data-answer-type=non-selected-correct-answer]',
+					),
+				}),
+		);
+
+		// Click the correct answer
+		await page.getByText('So that their poos don’t roll away').click();
+
+		// Click Reveal to show the results
+		await page
+			.locator('[data-atom-type=knowledgequiz]')
+			.getByText('Reveal')
+			.first()
+			.click();
+
+		// We were right!
+		await expectToExist(page, '[data-answer-type=correct-selected-answer]');
+	});
```

Comment:
Compare to the Cypress version:

https://github.com/guardian/dotcom-rendering/blob/394ba2a3802ca8374bf4f2cafe7ee7e411c40b78/dotcom-rendering/cypress/e2e/parallel-4/atom.interactivity.cy.js#L133-L155